smart/plugin.go

```diff
@@ -203,7 +205,9 @@ func (sc *SmartCollector) CollectMetrics(mts []plugin.MetricType) ([]plugin.Metr
 			for _, dev := range devices {
 				collected, result, err := sc.DiskMetrics(ns, t, dev, attribute_path, buffered_results, errs)
 				if err != nil {
-					sc.logger.Error(fmt.Sprintf("Error collecting SMART %s data on %s disk: %#+v", attribute_path, dev, err))
+					eStr := fmt.Sprintf("Error collecting SMART %s data on %s disk: %#+v", attribute_path, dev, err)
```

Comment:
Same as the comment below. It seems that it logs single error then append to log all errors together. We only need one way to log errors. I think. If the single error is more readable, stick to that.